cray_product_catalog/query.py

```diff
@@ -94,28 +96,26 @@ def __init__(self, name=PRODUCT_CATALOG_CONFIG_MAP_NAME, namespace=PRODUCT_CATAL
         self.namespace = namespace
         self.k8s_client = self._get_k8s_api()
         try:
-            config_map = self.k8s_client.read_namespaced_config_map(name, namespace)
+            configmaps = self.k8s_client.list_namespaced_config_map(
+                namespace, label_selector=PRODUCT_CATALOG_CONFIG_MAP_LABEL_STR
```

Comment:
Something occurred to me about this change. This change affects the timing of when we migrate `sat showrev` to use this new library for querying the product catalog.

If we update the version of `cray_product_catalog` used by `sat` too early, then `sat showrev` won't work on any older systems which have not had their `cray-product-catalog` ConfigMap migrated to the new one with the new label applied.

Would it be possible to make this code more backwards compatible? For example, if it doesn't find any configmaps with the `label_selector=PRODUCT_CATALOG_CONFIG_MAP_LABEL_STR`, then it could fall back on reading getting the single ConfigMap identified by `name` in `namespace` (i.e. `cray-product-catalog` in the `services` namespace). If it did this, then we could update the dependency in the `sat` code any time, and it should work both on older systems with the single `cray-product-catalog` ConfigMap and on newer systems with the split and labeled ConfigMaps.

On a related note, has anybody opened Jiras yet for updating other components which depend on this `cray-product-catalog` library. I think it would primarily be `product-deletion-utility`, `sat`, and `cfs-config-util`.

Reply:
@haasken-hpe Looks like this need discussion. Should only query need to be backward compatible or all features like update and delete?

Reply:
@lathanm, I think it would be pretty easy to at least make the query functionality backwards compatibility. If I recall correctly, both `sat` and `cfs-config-util` only use the query (read) functionality, and they don't make any updates to product catalog data. So if we provide backwards compatibility on the query/read functionality, it will make migration of those two a little easier.

I think that `product-deletion-utility` does delete from the product catalog, so if update/delete is not backwards compatible, it will just mean we need to be a little more careful to coordinate rolling out the changes to the `product-deletion-utility` at the same as the changes to this `cray-product-catalog` change.

Another component that depends on the `cray-product-catalog-update` image is the IUF. See here: https://github.com/Cray-HPE/docs-csm/blob/release/1.6/workflows/templates/base/update-product-catalog.template.argo.yaml

Am I missing any other dependencies we need to keep track of?

Perhaps you are already tracking all this work in Jiras and you have a plan for updating those other components which depend on this code. If you'd like to discuss further, I'd be happy to hop on a call. Thanks!